File: src/types/system/dto/windows/cpu.ts
export type IWindowsCpu = {
  Name: string; // OS에서 추출한 CPU 이름 (e.g, AMD Ryzen 7 9800X3D 8-Core Processor). DB에 저장되는 HW key 역할
  Manufacturer: string;
  Caption: string;
  NumberOfCores: number;
  NumberOfLogicalProcessors: number;
  MaxClockSpeed: number;
  CurrentClockSpeed: number | null;
  AddressWidth: number | null;
  Architecture: number | null;
  AssetTag: string | null;
  Availability: number | null;
  ProcessorId: string | null;
  Status: string | null;
};

File: src-tauri/src/system/windows/cpu.rs
use serde::{Deserialize, Serialize};
use wmi::{WMIConnection, WMIDateTime};

#[derive(Serialize, Deserialize, Debug)]
#[serde(rename = "Win32_Processor")]
#[serde(rename_all = "PascalCase")]
pub struct Win32Processor {
    address_width: Option<u16>,
    architecture: Option<u16>,
    asset_tag: Option<String>,
    availability: Option<u16>,
    caption: Option<String>,
    current_clock_speed: Option<u32>,
    manufacturer: Option<String>,
    max_clock_speed: Option<u32>,
    name: Option<String>,
    serial_number: Option<String>,
    number_of_cores: Option<u32>,
    number_of_logical_processors: Option<u32>,
    processor_id: Option<String>,
    status: Option<String>,
    // Add more fields here as needed
}

pub fn get_cpu_info(
    wmi_con: &WMIConnection,
) -> Result<Vec<Win32Processor>, Box<dyn std::error::Error>> {
    let processors: Vec<Win32Processor> = wmi_con.query()?;
    for processor in &processors {
        let processor_detail = format!("{:#?}\n", processor);
        println!("{}", processor_detail);
    }

    Ok(processors)
}